base/src/NvEglRenderer.cpp

```diff
@@ -152,21 +162,161 @@ NvEglRenderer::NvEglRenderer(const char *name, uint32_t width, uint32_t height,
+								                KeyPressMask |
+								                KeyReleaseMask |
+								                ButtonReleaseMask |
+								                EnterWindowMask |
+								                LeaveWindowMask |
+								                PointerMotionMask |
+								                PointerMotionHintMask |
+								                Button1MotionMask |
+								                Button2MotionMask |
+								                Button3MotionMask |
+								                Button4MotionMask |
+								                Button5MotionMask |
+								                ButtonMotionMask |
+								                KeymapStateMask |
+								                ExposureMask |
+								                VisibilityChangeMask |
+								                StructureNotifyMask |
+								                ResizeRedirectMask |
+								                SubstructureNotifyMask |
+								                SubstructureRedirectMask |
+								                FocusChangeMask |
+								                PropertyChangeMask |
+								                ColormapChangeMask |
+								                OwnerGrabButtonMask);
+
+        fontinfo = XLoadQueryFont(x_display, "9x15bold");
+
+        // XAllocNamedColor(x_display, DefaultColormap(x_display, screen_num), "green", &color, &dummy);
+        // XSetWindowBorder(x_display, x_window, color.pixel);
+
+        // gr_values.font = fontinfo->fid;
+        // gr_values.foreground = color.pixel;
+        // gr_values.line_width = 5;
+
+        // gc = XCreateGC(x_display, x_window, GCFont | GCForeground | GCLineWidth, &gr_values);
+
+        // XFlush(x_display);
+        // XMapWindow(x_display, (int32_t)x_window);
+        // XFlush(x_display);
+
+        XMapWindow(x_display, (int32_t)x_window);
+        gc = XCreateGC(x_display, x_window, 0, NULL);
+
+        XSetForeground(x_display, gc,
+                       WhitePixel(x_display, DefaultScreen(x_display)));
+        fontinfo = XLoadQueryFont(x_display, "9x15bold");
+        pthread_mutex_lock(&render_lock);
+        pthread_create(&render_thread, NULL, renderThread, this);
+        pthread_setname_np(render_thread, "EglRenderer");
+        pthread_cond_wait(&render_cond, &render_lock);
+        pthread_mutex_unlock(&render_lock);
 
-    XSetForeground(x_display, gc,
-                WhitePixel(x_display, DefaultScreen(x_display)) );
-    fontinfo = XLoadQueryFont(x_display, "9x15bold");
+        return;
+}
 
-    pthread_mutex_lock(&render_lock);
-    pthread_create(&render_thread, NULL, renderThread, this);
-    pthread_setname_np(render_thread, "EglRenderer");
-    pthread_cond_wait(&render_cond, &render_lock);
-    pthread_mutex_unlock(&render_lock);
+bool NvEglRenderer::renderAndDrawLoop()
+{
+    if (drawBorder)
+    {
+        XDrawRectangle(x_display, x_window, gc, 0, 0, (mWidth)-1, (mHeight)-1);
+        XFlush(x_display);
+    }
+    return true;
+}
 
-    return;
+bool NvEglRenderer::windowDrag()
+{
+    if (XCheckMaskEvent(x_display,
+                        ButtonPressMask |
```

Comment:
Option to enable/disable callback events

Reply:
create a separate issue if required.